### PR TITLE
Setup Smile Config Runtime [Alternative]

### PR DESCRIPTION
**Pull request by jumaallan**
Story: https://app.shortcut.com/smileid/story/xxx

## Summary

Vansh worked on an alternative implementation, so we can merge this into my open PR, then into develop after

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

### CODE REVIEW

File: sample/src/main/java/com/smileidentity/sample/activity/MainActivity.kt [hunk 1]
```diff
@@ -4,28 +4,17 @@ import android.os.Bundle
...
-                viewModel = viewModel(
-                    factory = viewModelFactory {
-                        MainScreenViewModel(
-                            isSmileIDInitialized = isSmileIDInitialized,
-                        )
-                    },
-                ),
-                isSmileIDInitialized = SmileID.isInitialized,
-            )
+            SmileIDTheme {
```
Comment (vanshg) [inline sample/src/main/java/com/smileidentity/sample/activity/MainActivity.kt:15]: The theme should be part of RootScreen? It would make the Previews more realistic

File: sample/src/main/java/com/smileidentity/sample/compose/components/SmileConfigModalBottomSheet.kt [hunk 2]
```diff
@@ -18,43 +18,33 @@ import androidx.compose.runtime.Composable
...
-    shouldShowSmileConfigBottomSheet: (shouldDismiss: Boolean) -> Unit,
-    updateSmileConfig: (updatedConfig: String) -> Boolean,
+    onSaveSmileConfig: (updatedConfig: String) -> Unit,
+    onDismiss: () -> Unit,
+    hint: String,
     modifier: Modifier = Modifier,
-    setSmileConfig: String = "",
-    smileConfig: Config? = null,
+    @StringRes errorMessage: Int? = null,
+    canDismissConfigSheet: Boolean = true,
```
Comment (vanshg) [inline sample/src/main/java/com/smileidentity/sample/compose/components/SmileConfigModalBottomSheet.kt:37]: nit: naming

`dismissable`

File: sample/src/main/java/com/smileidentity/sample/viewmodel/MainScreenViewModel.kt [hunk 3]
```diff
@@ -242,12 +224,8 @@ class MainScreenViewModel(
...
     }
 
-    fun showSmileConfigBottomSheet(shouldShowSmileConfigBottomSheet: Boolean) {
-        _uiState.update {
-            it.copy(
-                shouldShowSmileConfigBottomSheet = shouldShowSmileConfigBottomSheet,
-            )
-        }
+    fun showSmileConfigBottomSheet(showSmileConfigBottomSheet: Boolean) {
+        _uiState.update { it.copy(showSmileConfigBottomSheet = showSmileConfigBottomSheet) }
```
Comment (vanshg) [inline sample/src/main/java/com/smileidentity/sample/viewmodel/MainScreenViewModel.kt:228]: we don't need this anymore, right?

File: sample/src/main/java/com/smileidentity/sample/compose/RootScreen.kt
Comment (vanshg) [inline sample/src/main/java/com/smileidentity/sample/compose/RootScreen.kt:1]: You should look into how much of this is viable to be pushed into a viewmodel

**Review by vanshg — APPROVED** [review]

File: sample/src/main/java/com/smileidentity/sample/viewmodel/MainScreenViewModel.kt (hunk 3 above)
Comment (jumaallan) [reply sample/src/main/java/com/smileidentity/sample/viewmodel/MainScreenViewModel.kt:228]: Yeah, got rid of it

File: sample/src/main/java/com/smileidentity/sample/compose/RootScreen.kt
Comment (jumaallan) [reply sample/src/main/java/com/smileidentity/sample/compose/RootScreen.kt:1]: Fixed
